scripts/check_branch.py

```diff
@@ -21,9 +24,37 @@
 log = logging.getLogger(__name__)
 
 
-def default_deployment(branch: str | None) -> str | None:
+class BranchContext(Enum):
+    # The name of the base ref or target branch of the pull request in a
+    # workflow run. This is only set when the event that triggers a workflow
+    # run is either pull_request or pull_request_target. For example, main.
+    #
+    github_base = 'GITHUB_BASE_REF'
+    # The short ref name of the branch or tag that triggered the workflow
+    # run. This value matches the branch or tag name shown on GitHub. For
+    # example, feature-branch-1. For pull requests, the format is
+    # <pr_number>/merge.
+    #
+    github_ref = 'GITHUB_REF_NAME'
+    # The branch or tag name for which project is built.
+    #
+    gitlab = 'CI_COMMIT_REF_NAME'
```

Comment:
```suggestion
class BranchContext(Enum):
    # The name of the base ref or target branch of the pull request in a
    # workflow run. This is only set when the event that triggers a workflow
    # run is either pull_request or pull_request_target. For example, main.
    #
    github_pr = 'GITHUB_BASE_REF'
    
    # The short ref name of the branch or tag that triggered the workflow
    # run. This value matches the branch or tag name shown on GitHub. For
    # example, feature-branch-1. For pull requests, the format is
    # <pr_number>/merge.
    #
    github_push = 'GITHUB_REF_NAME'
    
    # The branch or tag name for which project is built.
    #
    gitlab = 'CI_COMMIT_REF_NAME'
```

Also, there is a preamble that attributes these comments to a source. The preamble would have to be moved as well.